### DISCUSSION TITLE
Domain filter rules access

### DISCUSSION THREAD

**Discussion by zomerstm** [Salamek/huawei-lte-api#143] [Ideas]
I'm not sure if this is the right place, but I'm using this API thru Home Assistant. IT would be nice, if there could be option, to turn off and on, created domain filter rules. Scenario - I'm blocking access to youtube in selected hours.

**Comment by Salamek** [comment]
This is more on HA implementation of this library... @scop

**Reply by scop** [reply]
True, eventually, but the basic building block is also missing from huawei-lte-api -- we only have support for retrieving the domain filter rules, not modifying/adding/removing them.

**Reply by zomerstm** [reply]
I think modifying/adding/removing is not necessary in API, is to complicated. But if there is support for retrieving it, it would be nice to get support for switching it on and off - thath should not be so complicated, I think ;)

After Salamek response, I also added feature request in HA forum: https://community.home-assistant.io/t/huawei-lte-switch-for-domain-filters/451417

**Comment by scop** [comment]
Nice use case. Although it's somewhat specific, and a more generic one would be to support per-rule control, I'm not sure if that on the other hand would be too fine grained. Also some concerns what to display as the filter name in HA, and what happens or not to it in HA if one goes and changes the domain in it, etc. But those concerns wouldn't exist if we'd just add support for turning all of them on or off.

Anyway as noted, huawei-lte-api lacks support for the operations necessary for this at the moment, so in that sense this is a good place to discuss it.

**Reply by Salamek** [reply]
Hmm i don't think that i have any device supporting this feature...

**Reply by zomerstm** [reply]
In my H122-373 11.0.2.11 - Advanced / Security / Domain name filter (translating from Polish).
Very strange, that it's not on other Huawei Devices, for me it's a very basic feature, even ma old cheap tp-link got it.

**Comment by scop** [comment]
https://github.com/Salamek/huawei-lte-api/pull/152 implements the setters required for this.
